File: examples/rest-api-python/src/update.py
import json
from db.notes import getNotes

def main(event, context):
  notes = getNotes()
  noteId = event["pathParameters"]["id"]

  if noteId in notes:
    eventBody = json.loads(event["body"])
    note = notes[noteId]
    note["content"] = eventBody["content"]
    return {
      "statusCode": 200,
      "body": json.dumps(note, indent=2)
    }

  return {
    "statusCode": 404,
    "body": json.dumps({
      "error": True
    })
  }

File: examples/rest-api/src/update.js
import notes from "./notes";

export async function main(event) {
  const note = notes[event.pathParameters.id];

  if (!note) {
    return {
      statusCode: 404,
      body: JSON.stringify({ error: true }),
    };
  }

  const data = JSON.parse(event.body);

  note.content = data.content;

  return {
    statusCode: 200,
    body: JSON.stringify(note),
  };
}
